platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -109,37 +114,38 @@ class HangingProtocolService {
     this.stage = undefined;
 
     this.studies = [];
-    Object.defineProperty(this, 'EVENTS', {
-      value: EVENTS,
-      writable: false,
-      enumerable: true,
-      configurable: false,
-    });
-    Object.assign(this, pubSubServiceInterface);
   }
 
-  public destroy() {
+  public destroy(): void {
     this.reset();
     this.protocols = new Map();
   }
 
-  public reset() {
+  public reset(): void {
     this.studies = [];
-    this.hpAlreadyApplied = new Map();
     this.viewportMatchDetails = new Map();
     this.displaySetMatchDetails = new Map();
   }
 
   /** Leave the hanging protocol in the initialized state */
-  public onModeExit() {
+  public onModeExit(): void {
     this.reset();
   }
 
+  /**
+   * Gets the active protocol information
+   * @returns protocol, stage, activeStudyUID
+   */
   public getActiveProtocol(): {
     protocol: HangingProtocol.Protocol;
     stage: number;
+    activeStudyUID?: string;
```

Comment:
this doesn't makes sense, it accepts `activeStudyUID` and returns the one from this?

Reply:
can we just use the `getState` for all of these? the only thing missing in the `getState` is the activeStudy that you can just add there.

Reply:
Switched to use getState for the state information and getActiveProtocol for the raw internal objects

Reply:
No, it does NOT take activeStudyUID - that is the response type value.